slack_bolt/adapter/aws_lambda/lazy_listener_runner.py

```diff
@@ -23,7 +23,7 @@ def start(self, function: Callable[..., None], request: BoltRequest) -> None:
         headers["x-slack-bolt-lazy-function-name"] = request.lazy_function_name  # not an array
         event["method"] = "NONE"
         invocation = self.lambda_client.invoke(
-            FunctionName=request.context["aws_lambda_function_name"],
+            FunctionName=request.context["aws_lambda_invoked_function_arn"],
```

Comment:
This should work. The lambda invocation API accepts any of function name, function ARN, and partial ARN. https://docs.aws.amazon.com/cli/latest/reference/lambda/invoke.html#options